platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/EmbeddedSolrFactory.java

```diff
@@ -46,167 +42,176 @@
 
   @Override
-  public Future<SolrClient> newClient(String core) {
-    ConfigurationStore configStore = ConfigurationStore.getInstance();
+  public org.codice.solr.client.solrj.SolrClient newClient(String core) {
+    final ConfigurationStore configStore = ConfigurationStore.getInstance();
 
     if (System.getProperty("solr.data.dir") != null) {
       configStore.setDataDirectoryPath(System.getProperty("solr.data.dir"));
     }
-
-    ConfigurationFileProxy configProxy = new ConfigurationFileProxy(configStore);
-    return Futures.immediateFuture(
-        getEmbeddedSolrServer(
-            core, HttpSolrClientFactory.DEFAULT_SOLRCONFIG_XML, null, configProxy));
-  }
-
-  /**
-   * Creates a new {@link EmbeddedSolrServer} using the default Solr core name ({@value
-   * #DEFAULT_EMBEDDED_CORE_NAME}), configuration file ({@value
-   * HttpSolrClientFactory#DEFAULT_SOLRCONFIG_XML}) and schema.
-   *
-   * @return a new {@link EmbeddedSolrServer} instance
-   */
-  public static SolrClient getEmbeddedSolrServer() {
-    return getEmbeddedSolrServer(
-        DEFAULT_EMBEDDED_CORE_NAME, HttpSolrClientFactory.DEFAULT_SOLRCONFIG_XML, null, null);
+    return newClient(
+        core,
+        HttpSolrClientFactory.DEFAULT_SOLRCONFIG_XML,
+        HttpSolrClientFactory.DEFAULT_SCHEMA_XML,
+        configStore,
+        new ConfigurationFileProxy(configStore));
   }
 
   /**
-   * Creates a new {@link EmbeddedSolrServer} using the Solr configuration file provided. Uses the
-   * default core name ({@value #DEFAULT_EMBEDDED_CORE_NAME}) and schema.
+   * Requests the creation of a new embedded {@code SolrClient} for a specific Solr core name.
    *
-   * @param solrConfigXml name of the Solr configuration file. Defaults to {@value
-   *     HttpSolrClientFactory#DEFAULT_SOLRCONFIG_XML} if {@code null}.
-   * @return a new {@link EmbeddedSolrServer} instance
+   * @param core the name of the Solr core to create to create a client for
+   * @param configXml the name of the Solr configuration file
+   * @param schemaXml the name of the Solr core schema
+   * @param configStore the {@link ConfigurationStore} instance to use
+   * @param configProxy the {@link ConfigurationFileProxy} instance to use
+   * @return the newly created {@code SolrClient}
+   * @throws IllegalArgumentException if <code>core</code>, <code>configXml</code>, <code>schemaXml
+   * </code>, <code>configStore</code> or <code>configProxy</code> is <code>null</code>
    */
-  public static EmbeddedSolrServer getEmbeddedSolrServer(@Nullable String solrConfigXml) {
-    return getEmbeddedSolrServer(DEFAULT_EMBEDDED_CORE_NAME, solrConfigXml, null, null);
+  public org.codice.solr.client.solrj.SolrClient newClient(
+      String core,
+      String configXml,
+      String schemaXml,
+      ConfigurationStore configStore,
+      ConfigurationFileProxy configProxy) {
+    Validate.notNull(core, "invalid null Solr core name");
```

Comment:
❓Should we also validate that the `String` arguments are not blank? For the two files, how will the caller know if they are invalid, i.e., can't be read, don't exist, etc.?

Reply:
First, I simply moved that code around to reduce cyclomatic complexity. Second, this is going away with the Solr as an external process effort. As such, I will leave it as is.